python/doc/examples/calibration/bayesian_calibration/plot_imh_python_distribution.py

```diff
@@ -1,100 +1,327 @@
+# the time with the generic sampler. The Ratio of Uniforms algorithm proves to be much more rapid
+# than the generic method.
+sizeRoU = 10000
+sizeGeneric = 100
+
+t0 = monotonic()
+sample_RoU = newDist_RoU.getSample(sizeRoU)
+t1 = monotonic()
+sample_generic = newDist_generic.getSample(sizeGeneric)
+t2 = monotonic()
+
+print("Ratio of Uniforms speed = ", sizeRoU / (t1 - t0), "rng/s")
+print("Generic algorithm speed = ", sizeGeneric / (t2 - t1), "rng/s")
+
+# %%
+# Case 2: Direct use the Ratio of Uniforms algorithm
+# --------------------------------------------------
+# In that case, we want to use the :class:`~openturns.experimental.RatioOfUniforms` algorithm to
+# sample :math:`p`. We need to compute the function :math:`\log f` and its range.
+#
+# We create the function :math:`\log f` and the :class:`~openturns.experimental.RatioOfUniforms`:
+log_UnscaledPDF = ot.ComposedFunction(ot.SymbolicFunction("x", "log(x)"), f)
+ratio_algo = otexp.RatioOfUniforms(log_UnscaledPDF, range_PDF)
+
+# %%
+# We can draw the exact PDF and the PDF estimated from a sample generated by the Ratio of Uniforms algorithm.
+
+# %%
+size = 100000
+sample = ratio_algo.getSample(size)
+
+ks_algo = ot.KernelSmoothing()
+ks_algo.setBoundaryCorrection(True)
+ks_algo.setLowerBound(lower_bound)
+ks_algo.setUpperBound(upper_bound)
+ks_pdf = ks_algo.build(sample)
+
+g = ks_pdf.drawPDF(-0.5, upper_bound * 1.1, 1001)
+g.add(draw_exact)
+g.setLegends(["RoU size = " + str(size), "exact PDF"])
+g.setLegendPosition("topright")
+g.setTitle("Ratio of Uniforms (RoU) ")
+g.setXTitle("x")
+view = otv.View(g)
+
+# %%
+# In order to facilitate the comparison with the use of Metropolis-Hastings
+# based algorithms, we generate
+# a sample of the same size and we draw the estimated PDF.
+size2 = 500
+sample = ratio_algo.getSample(size2)
+
+ks_algo = ot.KernelSmoothing()
+ks_algo.setBoundaryCorrection(True)
+ks_algo.setLowerBound(lower_bound)
+ks_algo.setUpperBound(upper_bound)
+ks_pdf_RoU = ks_algo.build(sample)
+
+draw_RoU = ks_pdf_RoU.drawPDF(-0.5, upper_bound * 1.1, 1001).getDrawable(0)
+draw_RoU.setLineWidth(2)
+g.add(draw_RoU)
+g.setLegends(
+    ["RoU size = " + str(size), "exact PDF", "RoU (size = " + str(size2) + ")"]
+)
+g.setTitle("Ratio of Uniforms")
+view = otv.View(g)
 
 # %%
-# Independent Metropolis-Hastings
-# -------------------------------
+# By default, the parameter :math:`r=1`. We can get the associated acceptance ratio:
+print("Acceptance ratio = ", ratio_algo.getAcceptanceRatio())
+
+# %%
+# We can estimate the normalization factor with the  Ratio of Uniforms algorithm (see
+# the documenttaion of :class:`~openturns.experimental.RatioOfUniforms`):
+print("Normalization factor estimate = ", ratio_algo.getC())
+
+# %%
+# We can change the :math:`r` parameter and check the associated acceptance ratio:
+r_new = 0.5
+ratio_algo.setR(r_new)
+print("New acceptance ratio = ", ratio_algo.getAcceptanceRatio())
+
```

Comment:
I am not sure to understand what we conclude from changing the parameter r from r=1 to r=0.5 considering the acceptance ratio changing from 0.100 to 0.111 ? 
Should we add a sentence to clarify the consequences of changing r here ?

Reply:
Yes I just wanted to show how to change the value of r....
I modified the text: I ploted the graph of $r \rightarrow \tau(r)$ to show that the default value $r=1$ is not so bad and that it is not worth changing its value.